projects/packages/post-list/src/class-admin.php

```diff
@@ -172,4 +171,39 @@ public function enqueue_scripts( $hook ) {
 			);
 		}
 	}
+
+	/**
+	 * Return the featured image or if no featured image is set, return the first image in the post. If neither exists
+	 * return the featured image array with null values.
+	 *
+	 * @param int $post_id The current post ID.
+	 * @return array The featured image id and URLs
+	 */
+	protected function get_featured_or_first_post_image( $post_id ) {
+		$featured_image_id    = null;
+		$featured_image_url   = null;
+		$featured_image_thumb = null;
+
+		if ( has_post_thumbnail() ) {
+			$featured_image_id    = get_post_thumbnail_id( $post_id );
+			$featured_image_url   = get_the_post_thumbnail_url( $post_id );
+			$featured_image_thumb = get_the_post_thumbnail_url( $post_id, array( 50, 50 ) );
+		} else {
+			// TODO: This is not working like I expected. I think if an image is on two posts, it only comes back as
+			// attached_media to the first post it was added to. Must rethink this.
+			$image = current( get_attached_media( 'image', $post_id ) );
```

Comment:
I spent some time investigating how to get the first image from a post if no featured image is set. I played around a bit with `get_attached_media( 'image', $post_id )`. That seemed to work at first, but in testing I'm using the same image in 2 posts and that image only appears as an "attached media" in the first post I added it to. Or at least, I think that's what is going on.

I also looked at Calypso a bit to see how it works there, but I haven't figured it out yet. 🤔

Reply:
Under further investigation. It appears that in regards to get_attached_media, an image is attached to the first post it is added to. That same image will not be returned by get_attached_media if it is included in any other posts. Further, if you add an image to a post, save it, then remove that image from the post, get_attached_media will still return that now removed image, because it was attached to that post first.

Due to these 2 issues, get_attached_media seems like a bad candidate for retrieving the first image from a post if a featured image is not available.

It may be best to use preg_match_all and a regular expression to get the first image from the post.

Reply:
maybe showing the image post as the featured image is not a good idea either. I wonder whether it could be confusing for the users.

Reply:
Are you suggesting we just use the featured image thumbnail without a fall back? I'm ok with that idea. However, personally when I use WordPress I never use the featured image function, and am just happily surprised when an image from my post shows up as a thumbnail. 😅 It is a sticky situation on which image to present though and how best to retrieve it.

Reply:
Whether only using the featured image or trying to inform the user that the thumbnail may not be the featured image but a post one, because in the end, they are different things? @lcollette could you weigh in here?

Reply:
> However, personally when I use WordPress I never use the featured image function, and am just happily surprised when an image from my post shows up as a thumbnail. 😅

This is why I prefer what we're doing in Calypso. I view the image as a visual reference to the post, rather than a feature. For this reason, I'm not too concerned with needing to inform the user of the differences. We can add a note to our documentation stating we default to the feature image, and fall back to the first image in the post.